shifthelper/checks/__init__.py

```diff
@@ -115,15 +115,27 @@ def __init__(self,
     def run(self):
         while not self.stop_event.is_set():
             if len(self.queue) > 0:
-                if self.caller is not None:
-                    self.caller.place_call()
+
                 while len(self.queue) > 0:
                     message = self.queue.popleft()
                     self.logger.warning(message)
```

Comment:
I'm not an expert in the Python threading model. Anyhow. The queue is shared across threads yes?
If so. This might be a race condition. Can these two lines be made atomic?